doc/notebooks/parallelization_tutorial.ipynb

```diff
@@ -0,0 +1,510 @@
+    "The following example shows IDW interpolation split into 4 tiles. In this case, specifying an overlap is needed to get correct results without edge artifacts. Number and size of tiles is derived from the number of cores, but can be specified."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8b45b4d1-2c72-4e9a-aeff-6f862dd860ba",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "g.region vector=elev_points res=1 -p"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7a46cfb0-e177-4ecf-8a1b-78a42d87b027",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%python\n",
+    "import time\n",
+    "from grass.pygrass.modules.grid import GridModule\n",
+    "import time\n",
+    "start = time.time()\n",
+    "grid = GridModule(\n",
+    "    \"v.surf.idw\",\n",
+    "    input=\"elev_points\",\n",
+    "    output=\"idw\",\n",
+    "    column=\"value\",\n",
+    "    npoints=25,\n",
+    "    processes=4,\n",
+    "    overlap=5,\n",
+    "    quiet=True,\n",
+    ")\n",
+    "grid.run()\n",
+    "print(f\"Elapsed time: {time.time() - start} s\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6d77dc60-68b9-478e-b458-9dbb2f36298c",
+   "metadata": {},
+   "source": [
+    "The following is the same tool ran in serial:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a7dfb34e-5147-4354-aba3-6cf7c144fc93",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "time v.surf.idw --q input=elev_points column=value output=idw npoints=25"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "953321f5-4745-4a23-94b3-5eeb3263db1f",
+   "metadata": {},
+   "source": [
+    "There are tools that already integrate tiling. For example, addon [r.mapcalc.tiled](https://grass.osgeo.org/grass-stable/manuals/addons/r.mapcalc.tiled.html) uses the tiling concept for  raster algebra computation. More complex algebra expression will increase the speedup of this method."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "48b78da4-f785-4447-be74-2cecf56c14f0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "g.extension r.mapcalc.tiled\n",
+    "g.region raster=elevation res=1\n",
+    "# parallel\n",
+    "time r.mapcalc.tiled expression=\"rescaled_elevation = graph(elevation,60,1,80,10,100,100,120,100,140,1000,160,1000)\" nprocs=4\n",
+    "# serial\n",
+    "time r.mapcalc expression=\"rescaled_elevation = graph(elevation,60,1,80,10,100,100,120,100,140,1000,160,1000)\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b3e920f3-1d53-4aad-b49b-31b447a352f0",
+   "metadata": {},
+   "source": [
+    "### Task-based parallelization\n",
+    "GRASS processing tools executed in parallel are separate processes that, unlike threads, do not share memory. When tasks are limited by disk I/O, parallel processing may have large overhead.\n"
```

Comment:
Connect to task-based p. Maybe "t-b are typically implemented by calling individual tools.